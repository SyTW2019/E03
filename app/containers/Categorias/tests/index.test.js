import React from 'react';
import ReactDOM from 'react-dom';
import { render, cleanup } from 'react-testing-library';
import "jest-dom/extend-expect";
import renderer from "react-test-renderer";

import Categoria from '../index'

const className = 'name';
const renderComponent = (props = {}) =>
  render(<Categoria className={name} {...props} />);


describe('<Categoria />', () => {

  afterEach(cleanup);

  it('render', () => {
      const div = document.createElement(div);
      ReactDOM.render(<Categoria/>,div);
  });

  it('should render an <h1> tag', () => {
    const { container } = renderComponent();
    const element = container.querySelector('h1');
    expect(element).not.toBeNull();
  });

  it('should adopt a className attribute', () => {
    const className = 'title';
    const { container } = renderComponent({ className });
    const element = container.querySelector('h1');
    expect(element.className).toEqual(className);
  });

  it('Matches snapshots', () => {
    const tree = renderer.create(<Categoria title="save"/>).toJSON();
    expect(tree).toMatchSnapshot();
  });


});