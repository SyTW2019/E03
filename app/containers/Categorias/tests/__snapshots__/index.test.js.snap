// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Categoria /> Matches snapshots 1`] = `
<h1
  className="title"
>
  Categor√≠as
</h1>
`;
